### PR TITLE
Add suggested TODO items

### PR DESCRIPTION
**Pull request by manopapad**
Included some of my own wishlist items, plus items from https://github.com/StanfordLegion/legion/issues/1551 and https://github.com/StanfordLegion/prof-viewer/issues/25

### CODE REVIEW

File: TODO.md [hunk 1]
```diff
@@ -24,6 +24,41 @@
...
+- [ ] Make highlighted boxes clearer
+
+  Currently a highlighted box is shown in red, but that color is already in use in the default color scheme, so highlighted items don't stand out
+
+  Possible solutions:
+
+  - "halo" effect for highlighted box
+  - different fill pattern
+  - different line color
+  - saturate all other boxes except the highlighted one (as happens for search)
```
Comment (elliottslaughter) [inline TODO.md:40]: Indent to line up with the first line of the bullet, so it gets parsed properly as Markdown.

File: TODO.md [hunk 2]
```diff
@@ -24,6 +24,41 @@
...
+- [ ] Provenance strings that contain a machine-readable part (after a `$` delimiter) are not parsed
+  
+  E.g. a full provenance string like:
+  
+      foo.py:42$OpCode,MULTIPLY|Provenance,foo.py:42
+
+  should be getting reported as:
+
+      foo.py:42
+      OpCode=MULTIPLY
```
Comment (elliottslaughter) [inline TODO.md:61]: This is fine but I actually want to make a proposal to change this format entirely.

File: TODO.md (hunk 1 above)
Comment (manopapad) [reply TODO.md:40]: Doing it this way

Screenshot 2023-09-27 at 5 31 11 PM

results in github misparsing it as a code block:

Screenshot 2023-09-27 at 5 31 23 PM

The original parses properly (at least by github):

Screenshot 2023-09-27 at 5 33 12 PM

File: TODO.md (hunk 2 above)
Comment (manopapad) [reply TODO.md]: ddc00d0

**Pull request merged**